src/main/java/com/bookstore/service/AuthorServiceDao.java

```diff
@@ -0,0 +1,97 @@
+package com.bookstore.service;
+
+import com.bookstore.database.DBConnection;
+import com.bookstore.model.Author;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AuthorServiceDao implements AuthorService {
+    @Override
+    public void addAuthor(Author author) {
+        try (Connection connection = DBConnection.getConnection()) {
+            try (PreparedStatement ps = connection.prepareStatement("INSERT INTO authors (id, first_name, last_name, email) VALUES (?, ?, ?, ?)")) {
+                ps.setInt(1, author.getId());
+                ps.setString(2, author.getFirstName());
+                ps.setString(3, author.getLastName());
+                ps.setString(4, author.getEmail());
+                ps.executeUpdate();
+            }
+        } catch (SQLException e) {
```

Comment:
since you are catching the exception your response will be 200 even if an exception was thrown. Same for some of the other function as well